File: kongsberg/node-rag-demo/build-index.mjs
// file: build-index.mjs
import 'dotenv/config';
import { OpenAIEmbeddings } from "@langchain/openai";
import { RecursiveCharacterTextSplitter } from "@langchain/textsplitters";
//import { RecursiveCharacterTextSplitter } from "langchain";
import { ChromaClient } from "chromadb";

const OPENAI_API_KEY = process.env.OPENAI_API_KEY;
const OPENAI_API_BASE = process.env.OPENAI_API_BASE;

async function main() {
  // 1. Sample docs (replace with loaders later)
  const rawDocs = [
    "Agentic AI tools can orchestrate multiple skills autonomously.",
    "RAG uses a retriever over a vector store plus an LLM generator.",
    "ChromaDB supports persistent and in‑memory vector collections.",
  ];

  const textSplitter = new RecursiveCharacterTextSplitter({
    chunkSize: 400,
    chunkOverlap: 50,
  });

  const splitDocs = [];
  for (const text of rawDocs) {
    const chunks = await textSplitter.splitText(text);
    for (const chunk of chunks) {
      splitDocs.push(chunk);
    }
  }

  // 2. Embeddings via LangChain
  const embeddings = new OpenAIEmbeddings({
    model: "text-embedding-3-large",
    openAIApiKey: OPENAI_API_KEY,
    configuration: {
      baseURL: OPENAI_API_BASE,
    },
  });

  const vectors = await embeddings.embedDocuments(splitDocs);

  // 3. Chroma client (using default in‑memory server; for prod, point to real host)
  //const client = new ChromaClient({
  //  path: "http://localhost:8000", // ensure chroma server is running here
  //});

  const client = new ChromaClient();

  const collection = await client.getOrCreateCollection({
    name: "agentic-ai-notes",
    metadata: { description: "Agentic AI training snippets" },
  });

  const ids = splitDocs.map((_, i) => `doc-${i}`);
  await collection.add({
    ids,
    documents: splitDocs,
    embeddings: vectors,
  });

  console.log("Chroma collection 'agentic-ai-notes' populated.");
}

main().catch(console.error);

File: kongsberg/py-rag-demo/build_index.py
# file: build_index.py
import os
from dotenv import load_dotenv

from langchain_openai import OpenAIEmbeddings
from langchain_chroma import Chroma
#from langchain.text_splitter import RecursiveCharacterTextSplitter
#from langchain.textsplitters import RecursiveCharacterTextSplitter
from langchain_text_splitters import RecursiveCharacterTextSplitter
from langchain_core.documents import Document

load_dotenv()

OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
OPENAI_API_BASE = os.getenv("OPENAI_API_BASE")

# 1. Your corpus (replace with file loaders in your training later)
raw_docs = [
    "Agentic AI systems can autonomously decide which tools to call.",
    "RAG combines retrieval from a vector store with generation from an LLM.",
    "Chroma is an open‑source vector database optimized for AI applications.",
]

# 2. Split into chunks
text_splitter = RecursiveCharacterTextSplitter(
    chunk_size=400,
    chunk_overlap=50,
)
docs = text_splitter.split_documents(
    [Document(page_content=t) for t in raw_docs]
)

# 3. Embeddings
embeddings = OpenAIEmbeddings(
    model="text-embedding-3-large",
    api_key=OPENAI_API_KEY,
    base_url=OPENAI_API_BASE,
)

# 4. Create / persist Chroma store
PERSIST_DIR = "./chroma-store"

vectorstore = Chroma.from_documents(
    documents=docs,
    embedding=embeddings,
    persist_directory=PERSIST_DIR,
)

print("Chroma index built at", PERSIST_DIR)
